test/e2e-setup/README.md

```diff
@@ -0,0 +1,65 @@
+# ARO HCP E2E Setup
+
+This set of scripts implements **setup of ARO HCP clusters in major
+configurations** for the needs of [ARO HCP E2E Tests](/test/e2e), OpenShift
+Cluster Validation Tests, as well as development and demonstration purposes of
+ARO HCP project.
+
+## Who can use ARO HCP E2E Setup and how?
+
+- ARO HCP subteams to create ARO HCP clusters in well known configurations
+  for development or testing purposes
+- ARO HCP QE subteam as a test setup for *ARO HCP E2E*
+  or *OpenShift Cluster Validation* test runs
+- ARO HCP subteams to refine and learn about existing
+  supported cluster configurations
+
+## In which environments should ARO HCP E2E Setup work?
+
+We plan for the setup to work in [all environments](/docs/environments.md):
+
+- [ARO HCP Dev enviroment](https://github.com/Azure/ARO-HCP/blob/main/dev-infrastructure/docs/development-setup.md)
+  for the needs of ARO HCP teams in Red Hat (no matter if a shared integrated
+  DEV environment or personal dev environment is used)
+- Internal integration and stage environments of ARO HCP team
+- Azure Production (in other words, one should be able to use it with
+  ARO HCP production service as any customer)
+
+That said right now, Dev env. is not supported. This is a temporary limitation.
+
+## Structure and Design Choices
+
+### Bicep
+
+[Bicep](https://learn.microsoft.com/en-us/azure/azure-resource-manager/bicep/overview?tabs=bicep)
+is used a primary way to describe ARO HCP cluster configurations for E2E Setup.
+
+A cluster configuration (to deploy via E2E setup) is defined in a single
+bicep file in `test/e2e-setup/bicep` directory. For example:
+
+* [`demo.bicep`](https://github.com/Azure/ARO-HCP/blob/main/test/e2e-setup/bicep/demo.bicep) is a simple configuration based on the original demo,
+* [`infra-only.bicep`](https://github.com/Azure/ARO-HCP/blob/main/test/e2e-setup/bicep/infra-only.bicep) is a setup which defines just customer deployed infra
+  (like network, managed identities ...) without creating ARO HCP cluster.
```

Comment:
Need to add the bicep template that provides the managed identities here.

Reply:
No need for that. The bicep modules provisioning particular components are not expected to be used directly, but only via bicep files in the `/test/e2e-setup/bicep/` directory, such as the examples listed above.

So if you need just deploy customer infra including managed identities, you just use `infra-only.bicep`. The outputs of this template explicitly include the MI structures (even though, you can access it via the managed identity template directly as well).